core/async/src/test_loop/delay_sender.rs

```diff
@@ -0,0 +1,69 @@
+use std::{sync::Arc, time::Duration};
+
+use crate::messaging;
+
+use super::multi_instance::IndexedDelaySender;
+
+/// Interface to send an event with a delay (in virtual time). It can be
+/// converted to a Sender for any message type that can be converted into
+/// the event type, so that a DelaySender given by the test loop may be passed
+/// to production code that expects a Sender.
+pub struct DelaySender<Event> {
+    // We use an impl object here because it makes the interop with other
+    // traits much easier.
+    im: Arc<dyn CanSendWithDelay<Event>>,
```

Comment:
AFAIK the convention for introducing isomorphic types, is to either
* make the struct an unnamed tuple (in your case: struct DelaySender<Event>(CanSendWithDelay<Event>))
* call the wrapped field "inner".

Reply:
I am actually taking one of your suggestions on the other PR where you suggested to define Sender as a wrapper of a function. Here this actually makes a lot of sense because we control all implementations of DelaySender. It gets rid of quite a bit of boilerplate.